packages/repository/src/legacy-juggler-bridge.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export const jugglerModule = require('loopback-datasource-juggler');
+
+import {MixinBuilder} from './mixin';
+import {Class, ObjectType, Options, AnyType} from './common';
+
+import {juggler} from './loopback-datasource-juggler';
+
+export * from './loopback-datasource-juggler';
+
+/* tslint:disable-next-line:variable-name */
+export const DataSource = jugglerModule.DataSource as typeof juggler.DataSource;
+/* tslint:disable-next-line:variable-name */
+export const ModelBase = jugglerModule.ModelBaseClass as typeof juggler.ModelBase;
+
+/**
+ * This is a bridge to the legacy DAO class. The function mixes DAO methods
+ * into a model class and attach it to a given data source
+ * @param modelClass {} Model class
+ * @param ds {DataSource} Data source
+ * @returns {} The new model class with DAO (CRUD) operations
+ */
+export function bindModel<T extends typeof juggler.ModelBase>(modelClass: T,
+  ds: juggler.DataSource): T {
+  const boundModelClass = class extends modelClass {};
+  boundModelClass.attachTo(ds);
+  return boundModelClass;
+}
+
+import {Entity} from './model';
+import {Filter, Where} from './query';
+import {EntityCrudRepository} from './repository';
+
+function isPromise<T>(p: AnyType): p is Promise<T> {
```

Comment:
Cross-posting my earlier comment:

The function is called `getPromise` but it either returns the argument value or an error (via a promise rejection). That's not the behaviour I'd expect from a getter method. I am proposing to rename the method to something like `ensurePromise`.